pdata/pprofile/attributes.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package pprofile // import "go.opentelemetry.io/collector/pdata/pprofile"
+
+import (
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+type attributable interface {
+	AttributeIndices() pcommon.Int32Slice
+}
+
+// BuildAttributes builds a [pcommon.Map] containing the attributes of a record.
+// The record can by any struct that implements an `AttributeIndices` method.
+func BuildAttributes(profile Profile, record attributable) pcommon.Map {
```

Comment:
Should this be a func on `Profile` then?

Reply:
I don't mind moving this into Profile, but mixing generated code within non-generated usually tends to harder to read code.
I've switched to passing only the `AttributeTable`.